Makefile

```diff
@@ -272,16 +256,33 @@ mocks: $(GOPATH)/bin/mockery
 .PHONY: codegen
 codegen: status proto swagger mocks docs
 
-.PHONY: proto
-proto:
-	$(call backup_go_mod)
-	# We need the folder for compatibility
+# you cannot install a specific version using `go install`, so we do this business
+.PHONY: tools
+tools:
 	go mod vendor
-	# Generate proto
+	go install ./vendor/github.com/go-swagger/go-swagger/cmd/swagger
+	go install ./vendor/github.com/gogo/protobuf/protoc-gen-gogo
+	go install ./vendor/github.com/gogo/protobuf/protoc-gen-gogofast
+	go install ./vendor/github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway
+	go install ./vendor/github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger
+	go install ./vendor/k8s.io/code-generator/cmd/go-to-protobuf
+	go install ./vendor/k8s.io/kube-openapi/cmd/openapi-gen
+	rm -Rf vendor
+
+$(GOPATH)/bin/go-to-protobuf: tools
+$(GOPATH)/bin/protoc-gen-gogo: tools
+$(GOPATH)/bin/protoc-gen-gogofast: tools
+$(GOPATH)/bin/protoc-gen-grpc-gateway: tools
+$(GOPATH)/bin/protoc-gen-swagger: tools
+$(GOPATH)/bin/swagger: tools
```

Comment:
Would it not be better to replace individual binary names with `tools`? I.e. cut out the middle man

Reply:
how strongly do you feel? I like that the indirection tells me what the actual dependency is rather than "i have dependency on tools, but I won't tell you what"